src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -438,53 +503,130 @@ void GfxBatchHbaoTest::testFlippedHBAOData(const TestDataType& data,
 
 void GfxBatchHbaoTest::testPerspectiveFlipped() {
   auto&& data = TestData[testCaseInstanceId()];
   setTestCaseDescription(
-      Cr::Utility::formatString("{}, perspective, flipped", data.name));
+      Cr::Utility::format("{}, perspective, flipped", data.name));
   testFlippedHBAOData(
-      data,
-      Cr::Utility::formatString("{}.{}.png", baseTestFilename, data.filename),
+      data, Cr::Utility::format("{}.{}.png", baseTestFilename, data.filename),
       flippedPerspectiveData);
 
 }  // GfxBatchHbaoTest::testPerspective()
 
-// void GfxBatchHbaoTest::benchmarkPerspective() {
-//   auto&& data = TestData[testCaseInstanceId()];
-//   setTestCaseDescription(
-//       Cr::Utility::formatString("{}, perspective benchmark.", data.name));
+void GfxBatchHbaoTest::benchmarkHBAOData(const TestDataType& data,
+                                         Mn::Matrix4 projMatrix) {
+  if ((data.config.flags() & esp::gfx_batch::HbaoFlag::LayeredImageLoadStore) &&
+      !(
+#ifdef MAGNUM_TARGET_GLES
+          Mn::GL::Context::current().isVersionSupported(
+              Mn::GL::Version::GLES310)
+#else
+          Mn::GL::Context::current()
+              .isExtensionSupported<
+                  Mn::GL::Extensions::ARB::shader_image_load_store>()
+#endif
+              ))
+    CORRADE_SKIP("Image load/store not supported");
+
+  // Scale image size to use for benchmark uniformly so projection matrix aspect
+  // ration not affected.
+  Mn::Vector2i BenchImageSize = Size * 4;
+  const size_t arraySize = BenchImageSize.x() * BenchImageSize.y() * 8;
+
+  // For benchmarks source color and depth are just empty white images
+
+  Cr::Containers::Array<char> depthData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D depth{Mn::PixelFormat::Depth32F, BenchImageSize,
+                               std::move(depthData)};
+
+  Mn::GL::Texture2D inputDepthTexture;
+  inputDepthTexture
+      .setStorage(1, Mn::GL::TextureFormat::DepthComponent32F, depth.size())
+      .setSubImage(0, {}, depth);
+
+  Cr::Containers::Array<char> colorData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D color{Mn::PixelFormat::RGBA8Unorm, BenchImageSize,
+                               std::move(colorData)};
+  Mn::GL::Texture2D outputColorTexture;
+  outputColorTexture.setStorage(1, Mn::GL::TextureFormat::RGBA8, color.size())
+      .setSubImage(0, {}, color);
+
+  Cr::Containers::Array<char> resColorData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D resultImage{Mn::PixelFormat::RGBA8Unorm,
+                                     BenchImageSize, std::move(resColorData)};
+
+  Mn::GL::Framebuffer output{{{}, BenchImageSize}};
+  output.attachTexture(Mn::GL::Framebuffer::ColorAttachment{0},
+                       outputColorTexture, 0);
+
+  /* No clear, that would kill the base image */
+
+  MAGNUM_VERIFY_NO_GL_ERROR();
+
+  auto config{data.config};
+  esp::gfx_batch::Hbao hbao{config.setSize(BenchImageSize)};
```

Comment:
```suggestion
  esp::gfx_batch::Hbao hbao{esp::gfx_batch::HbaoConfiguration{config}.setSize(BenchImageSize)};
```

Is a terser way to do the same, if you feel that's still readable.